Sources/MongoSwiftSync/MongoCursor.swift

```diff
@@ -1,58 +1,126 @@
 import MongoSwift
 
-/// A MongoDB cursor.
-public class MongoCursor<T: Codable>: Sequence, IteratorProtocol {
-    /// The error that occurred while iterating this cursor, if one exists. This should be used to check for errors
-    /// after `next()` returns `nil`.
-    public var error: Error? { fatalError("unimplemented") }
+/**
+ * A MongoDB cursor.
+ *
+ * Note that the `next` method blocks until a result is received or the cursor is exhausted, so methods inherited from
```

Comment:
As of right now, we're progressing forward in this PR by conforming everything to `LazySequenceProtocol` instead of `Sequence`, though I think a `tail` approach could be something worth exploring too.